### PR TITLE
[Unique-10] ConvertPrecision transformation and new attribute for output type

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 - ConvertPrecision transformation for Unique-10 op
 - New attribute to Unique-10 op to set the last output type 
   -> Specification update
   -> Python API update
   -> Core op update
   -> Additional tests

 

### Tickets:
 - related to 26815, 93675

### CODE REVIEW

File: docs/ops/movement/Unique_10.md [hunk 1]
```diff
@@ -74,7 +82,7 @@ The operator can either work in elementwise mode searching for unique values in
             <dim>-1</dim>
         </port>
         <port id="4" precision="I32">
-            <dim>-1</dim>
+            <dim>3</dim>
```
Comment (tomdol) [inline docs/ops/movement/Unique_10.md:85]: I think it's very unlikely that just this output gets a static shape while the others are still dynamic. Even though this is just an example I think it should reflect the behavior of the shape inference of this operation.

File: src/common/transformations/tests/utils/convert_precision.cpp [hunk 2]
```diff
@@ -323,6 +324,37 @@ TEST(TransformationTests, ConvertPrecision_TopK) {
...
+    std::shared_ptr<ov::Model> model(nullptr);
+    {
+        auto input = std::make_shared<op::Parameter>(element::f16, Shape{15, 20, 3});
+        auto unique = std::make_shared<opset10::Unique>(input);
+
+        model = std::make_shared<ov::Model>(unique->outputs(), ParameterVector{input});
+
+        pass::Manager manager;
+
+        static const precisions_array precisions = {
```
Comment (tomdol) [inline src/common/transformations/tests/utils/convert_precision.cpp:337]: Missing `auto`?

File: docs/ops/movement/Unique_10.md (hunk 1 above)
Comment (mitruska) [reply docs/ops/movement/Unique_10.md:85]: Such improvement was already introduced with reference implementation for Unique.
Please take a look at your changes in the type_prop tests:
https://github.com/openvinotoolkit/openvino/pull/13945/files#diff-18bd27a012b759129bbab5d44c8aa765cf32f76a8d9430bc38026414d81d2171R39
So this update is just to align the spec with the shape inference and reflect current behavior.

File: src/common/transformations/src/transformations/convert_precision.cpp [hunk 3]
```diff
@@ -335,6 +338,20 @@ bool fuse_type_to_random_uniform_v8(const std::shared_ptr<ngraph::Node>& node, o
     return false;
 }
 
+bool fuse_type_to_unique_v10(const std::shared_ptr<Node>& node, ov::element::Type to, size_t idx) {
+    if (auto unique = ov::as_type_ptr<opset10::Unique>(node)) {
+        if (to == ov::element::i32 || to == ov::element::i64) {
+            if (idx == 1 || idx == 2) {
+                unique->set_index_element_type(to);
+            } else if (idx == 3) {
+                unique->set_count_element_type(to);
```
Comment (itikhono) [inline src/common/transformations/src/transformations/convert_precision.cpp:347]: it looks like we need to `return true` if we set a new type

Comment (mitruska) [reply src/common/transformations/src/transformations/convert_precision.cpp:347]: I thought that too, but the conversion is not called for each output when `fuse_to_...` function returns `true` and it looks like `true` value should be used when `replace_node` is used:
https://github.com/openvinotoolkit/openvino/blob/857ba7506ecadec2eacde6e842c725ad58aebbcd/src/common/transformations/src/transformations/convert_precision.cpp#L144-L146
Other way `return true` breaks the loop and the loop in `convert_node_output_precision` doesn't iterate over all of the outputs.
https://github.com/openvinotoolkit/openvino/blob/857ba7506ecadec2eacde6e842c725ad58aebbcd/src/common/transformations/src/transformations/convert_precision.cpp#L133

Another solution is to call `set_index_element_type` and `set_count_element_type` regardless the `idx` variable, so for all of the outputs at once. Then return `true` can be used.

File: src/common/transformations/tests/utils/convert_precision.cpp (hunk 2 above)
Comment (mitruska) [reply src/common/transformations/tests/utils/convert_precision.cpp:337]: Nope, `precisions_array` is the type.
https://github.com/openvinotoolkit/openvino/blob/823ea7c68a04bbbac74cc701375ec1f2c08aff7e/src/common/transformations/include/transformations/convert_precision.hpp#L74

**Review by tomdol — APPROVED** [review]

**Review by nshchego — APPROVED** [review]

File: src/common/transformations/src/transformations/convert_precision.cpp (hunk 3 above)
Comment (mitruska) [reply src/common/transformations/src/transformations/convert_precision.cpp:347]: Discussed offline, the adjustment of `convert_node_output_precision` will be added in a separate PR.
cc @tomdol

**Pull request merged**